src/repos/UserSessionRepo.ts

```diff
@@ -0,0 +1,139 @@
+import { getConnectionManager, Repository, getRepository } from 'typeorm';
+import { LoginTicket } from 'google-auth-library/build/src/auth/loginticket';
+import { SerializedUserSession } from '../common/types';
+
+import AppDevUtils from '../appdev/AppDevUtils';
+import UserRepo from './UserRepo';
+import User from '../entities/User';
+import UserSession from '../entities/UserSession';
+
+const db = (): Repository<UserSession> =>
+  getConnectionManager()
+    .get()
+    .getRepository(UserSession);
+
+/**
+ * Create or update session for a user
+ * @param {User} user - User to either create or update a session for
+ * @param {string | undefined} accessToken - Access token to be used for the session
+ * @return {UserSession} New or updated session
+ */
+const createOrUpdateSession = async (
+  user: User,
+  accessToken?: string
+): Promise<UserSession> => {
+  let session = await db()
+    .createQueryBuilder('usersessions')
+    .innerJoin('usersessions.user', 'user', 'user.uuid = :userID')
+    .setParameters({ userID: user.id })
+    .getOne();
+  if (session) {
+    session.update(accessToken);
+    session.user = user;
+  } else {
+    session = UserSession.fromUser(user, accessToken);
+  }
+
+  return db().save(session);
+};
+
+/**
+ * Creates a user if one doesn't exist and then initializes a session for them
+ * @param {LoginTicket} login - Login object provided by Google
+ * @return {SerializedUserSession} Contains session information for user
+ */
+const createUserAndInitializeSession = async (
+  login: LoginTicket
+): Promise<SerializedUserSession> => {
+  const payload = login.getPayload();
+
+  if (!payload) {
+    throw Error('Unable to login with Google');
+  }
+
+  const googleID = payload.sub;
+  const first = payload.given_name ? payload.given_name : '';
+  const last = payload.family_name ? payload.family_name : '';
```

Comment:
I believe javascript has some nifty syntax where you could just do `payload.given_name || ''` and `payload.family_name || ''`